src/containerapp/azext_containerapp/_utils.py

```diff
@@ -497,6 +516,171 @@ def validate_custom_location(cmd, custom_location=None):
+    try:
+        connected_cluster = get_connected_k8s(cmd, connected_cluster_id=connected_cluster_id)
+    except Exception as e:
+        handle_non_404_status_code_exception(e)
+    if connected_cluster is None:
+        raise ResourceNotFoundError("Cannot find connected cluster with connected cluster ID {}".format(connected_cluster_id))
+
+
+def get_connected_k8s(cmd, connected_cluster_id=None):
+    parsed_connected_cluster = parse_resource_id(connected_cluster_id)
+    subscription = parsed_connected_cluster.get("subscription")
+    cluster_rg = parsed_connected_cluster.get("resource_group")
+    cluster_name = parsed_connected_cluster.get("name")
+    return connected_k8s_client_factory(cmd.cli_ctx, subscription_id=subscription).get(resource_group_name=cluster_rg, cluster_name=cluster_name)
+
+
+def list_cluster_extensions(cmd, connected_cluster_id):
+    parsed_extension = parse_resource_id(connected_cluster_id)
+
+    subscription_id = parsed_extension.get("subscription")
+    cluster_rg = parsed_extension.get("resource_group")
+    cluster_rp = parsed_extension.get("namespace")
+    cluster_type = parsed_extension.get("type")
+    cluster_name = parsed_extension.get("name")
+    extension_list = k8s_extension_client_factory(cmd.cli_ctx, subscription_id=subscription_id).list(
+        resource_group_name=cluster_rg,
+        cluster_rp=cluster_rp,
+        cluster_resource_name=cluster_type,
+        cluster_name=cluster_name)
+    return extension_list
+
+
+def create_extension(cmd, extension_name, connected_cluster_id=None, namespace=None, logs_customer_id=None, logs_share_key=None, location=None, logs_rg=None):
+    from base64 import b64encode
+    from azure.cli.core.commands import LongRunningOperation
+    from azext_containerapp.vendored_sdks.kubernetesconfiguration import models
+
+    if logs_customer_id is None or logs_share_key is None:
+        logs_customer_id, logs_share_key = _generate_log_analytics_if_not_provided(cmd, logs_customer_id,
+                                                                                   logs_share_key, location, logs_rg)
+
+    parsed_extension = parse_resource_id(connected_cluster_id)
+    subscription = parsed_extension.get("subscription")
+    cluster_rg = parsed_extension.get("resource_group")
+    cluster_rp = parsed_extension.get("namespace")
+    cluster_type = parsed_extension.get("type")
+    cluster_name = parsed_extension.get("name")
+
+    e = models.Extension()
+    e.identity = models.Identity(type="SystemAssigned")
+    e.extension_type = CONTAINER_APP_EXTENSION_TYPE
+    e.release_train = 'stable'
+    e.auto_upgrade_minor_version = True
+
+    e.scope = models.Scope(cluster=models.ScopeCluster(release_namespace=namespace))
+
+    e.configuration_settings = {
+        "Microsoft.CustomLocation.ServiceAccount": "default",
+        "appsNamespace": namespace,
+        "clusterName": extension_name,
+        "logProcessor.appLogs.destination": "log-analytics"
+    }
```

Comment:
log analytics is required?

Reply:
Not required, but recommend.

Reply:
It is consistent with managed environment.